flink-formats/flink-compress/src/main/java/org/apache/flink/formats/compress/CompressWriterFactory.java

```diff
@@ -77,4 +90,16 @@ public String codecExtension() {
 			: "";
 	}
 
+	private void initializeCompressionCodec() {
```

Comment:
the configuration its only used to obtain the CompressCodec, we can use the new Configuration() and avoid this

Reply:
I believe this is fine. We are also loading the config from the map, if a Hadoop configuration was provided. It makes sense to group them under a single function.